python/Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py

```diff
@@ -241,6 +241,9 @@ def test_h_testXMLIndex(self):
             index_cat = raw_j._category
             this_index_cache = (index_cat, index_cls, index_cache)
 
+            print("just-built index: %s" % str(this_index_cache))
+            print("from disk: %s" % str(obj))
```

Comment:
Yes this is needed:
```
52rcurrie@lxplus047:->py.test Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py
============================================================================== test session starts ==============================================================================
platform linux2 -- Python 2.7.9, pytest-2.8.7, py-1.4.31, pluggy-0.3.1 -- /afs/cern.ch/sw/lcg/releases/LCG_79/Python/2.7.9.p1/x86_64-slc6-gcc49-opt/bin/python
cachedir: ../.cache
rootdir: /afs/cern.ch/user/r/rcurrie/cmtuser/GANGA/GANGA_Dev/install/ganga, inifile: setup.cfg
collected 9 items

Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_a_JobConstruction PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_b_JobXMLExists PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_c_XMLAutoUpdated PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_d_XMLUpdated PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_e_SubJobXMLExists PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_f_testXMLContent PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_g_testSJXMLContent PASSED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_h_testXMLIndex FAILED
Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py::TestSJXMLGenAndLoad::test_i_testSJXMLIndex PASSED

=================================================================================== FAILURES ====================================================================================
____________________________________________________________________ TestSJXMLGenAndLoad.test_h_testXMLIndex ____________________________________________________________________

self = <GPI.newXMLTest.TestSJXMLGenAndLoad.TestSJXMLGenAndLoad testMethod=test_h_testXMLIndex>

    def test_h_testXMLIndex(self):
        # Check index of job
        from Ganga.Core.GangaRepository.PickleStreamer import to_file, from_file

        from Ganga.GPI import jobs

        j = jobs(0)

        assert path.isfile(getIndexFile(j))

        with open(getIndexFile(j)) as handler:
            obj, errs = from_file(handler)

            assert isinstance(obj, tuple)

            from Ganga.GPIDev.Base.Proxy import stripProxy, getName
            raw_j = stripProxy(j)
            index_cache = raw_j._getRegistry().getIndexCache(raw_j)
            assert isinstance(index_cache, dict)

            index_cls = getName(raw_j)
            index_cat = raw_j._category
            this_index_cache = (index_cat, index_cls, index_cache)

>           print("just-built index: %s" % this_index_cache)
E           TypeError: not all arguments converted during string formatting

Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py:244: TypeError
```

Either we submit a bug report to py.test or wherever else is breaking part of the language or we translate this into a string. Otherwise the test __will__ fail.